be/src/serde/column_array_serde.cpp

```diff
@@ -53,41 +58,152 @@ const uint8_t* read_raw(const uint8_t* buff, void* target, size_t size) {
     return buff + size;
 }
 
-template <typename T>
+// only support sorted 32-bit integers
+template <bool sorted_32ints>
+uint8_t* encode_integers(const void* data, size_t size, uint8_t* buff, int encode_level) {
+    if (encode_level == 0) {
+        throw std::runtime_error("integer encode level does not work.");
+    }
+    uint64_t encode_size = 0;
+    if (sorted_32ints) {
+        encode_size = streamvbyte_delta_encode(reinterpret_cast<const uint32_t*>(data), (3 + size) * 1.0 / 4.0,
+                                               buff + sizeof(uint64_t), 0);
+    } else {
+        encode_size = streamvbyte_encode(reinterpret_cast<const uint32_t*>(data), (3 + size) * 1.0 / 4.0,
+                                         buff + sizeof(uint64_t));
+    }
+    buff = write_little_endian_64(encode_size, buff);
+
+    VLOG_ROW << fmt::format("raw size = {}, encoded size = {}, integers compression ratio = {}\n", size, encode_size,
+                            encode_size * 1.0 / size);
+    return buff + encode_size;
+}
+
+template <bool sorted_32ints>
+const uint8_t* decode_integers(const uint8_t* buff, void* target, size_t size) {
+    uint64_t encode_size = 0;
+    buff = read_little_endian_64(buff, &encode_size);
+    uint64_t encode_size1 = 0;
+    if (sorted_32ints) {
+        encode_size1 = streamvbyte_delta_decode(buff, (uint32_t*)target, (3 + size) * 1.0 / 4.0, 0);
+    } else {
+        encode_size1 = streamvbyte_decode(buff, (uint32_t*)target, (3 + size) * 1.0 / 4.0);
+    }
+    if (encode_size != encode_size1) {
+        throw std::runtime_error(fmt::format(
```

Comment:
why not return error status, but throw runtime error? will we meet these runtime_errors in user prod envs?

Reply:
that will change return type of some functions, related a lot of codes.

if this error happens, it is fatal error, we should close the encoding function.